src/lib/pickles/wrap_verifier.ml

```diff
@@ -369,8 +369,20 @@ struct
           (* TODO: num_bits should maybe be input_length - 1. *)
           Ops.bits_per_chunk * Ops.chunks_needed ~num_bits:input_length
         in
-        let rec pow2pow x i =
-          if i = 0 then x else pow2pow Inner_curve.Constant.(x + x) (i - 1)
+        (* computes 2^i *)
```

Comment:
I don't understand this change

Reply:
Before I found the issue with `new_unchecked` this removed most of the regression by replacing a lot of individual calls to `add` by much less calls to `scale`. I'm checking now if this `pow2pow` is faster after the `new_unchecked` fix, if not I can revert.

Reply:
Originally I did it because addition was much more expensive than scaling due to `new_unchecked` regression. 

But I just ran the test after regression (https://github.com/MinaProtocol/mina/commit/92c1e79e2af76498a1b5627c73e6cb4b38c8ddd3) and the results are that doing one scaling is still about 30 cheaper than many additions:

```
test_pow2pow old: 15.285044 secs
test_pow2pow new: 0.511066 secs
```

Reply:
I think this is because `Constant.(+)` is implemented with a lot of extra checks:

https://github.com/MinaProtocol/mina/blob/volhovm/arkworks042-compatible-backup/src/lib/pickles/step_main_inputs.ml#L157

(I'm not sure if they're necessary)

Reply:
`Constant.(+)` is still slower even without zero checks. I would assume the most time is spent converting from projective and back. Basically, `Constant` is an affine representation, but it relies on projective bindings, so every time a function is called, points are first converted to projective, then operation (addition, negation) is done, and then it's converted back.

Reply:
LGTM. Thanks for the check. Useful.